### ISSUE TITLE
Environment Variable to Drupal configuration mapping magic

### ISSUE THREAD

**Issue by seanhamlin** [amazeeio/lagoon#1884]
At the moment, we are going through changes to the Fastly module to better accomodate some of the things we need to run it at scale inside Lagoon.

There has been discussion on how some of the configuration of the module should be supplied, and whether environment variables are the right way to go here.

Being able to seed Drupal 7's `$conf` variable and Drupal 8/9's `$config` and `$settings` variable with the values from environment variables would be ace.

## Prior art

From [Platform.sh docs](https://docs.platform.sh/development/variables.html#drupal-specific-variables):

> As a convention, our provided Drupal template code will automatically map variables to Drupal’s configuration system. The logic varies slightly depending on the Drupal version.
> 
> On Drupal 7 , any variable that begins with `drupal:` will be mapped to the global `$conf` array, which overrides Drupal’s `variable_get()` system. For instance, to force a site name from the Platform.sh variables (say to set it “This is a Dev site”) you would set the `drupal:site_name` variable.
> 
> On Drupal 8 , any variable that begins with `drupal:` will be mapped to the global `$settings` array. That is intended for very low-level configuration.
> 
> Also on Drupal 8, any variable that begins with `d8config:` will be mapped to the global `$config` array, which is useful for overriding drupal’s exportable configuration system. The variable name will need to contain two colons, one for d8config: and one for the name of the configuration object to override. For example, a variable named `d8config:system.site:name` will override the name property of the `system.site` configuration object.
> 
> As the above logic is defined in a file in your Git repository you are free to change it if desired. The same behavior can also be easily implemented for any other application or framework.

**Comment by Schnitzel** [comment]
I think think this is a great idea! As Lagoon is not forcefully injecting any php files into the application (and hopefully never will do), the only place for this would be in the example settings.php files: https://github.com/amazeeio/lagoon/blob/master/docs/using_lagoon/drupal/drupal8-composer-mariadb/web/sites/default/settings.php to implement some PHP code that iterates though all environment variables and if they start with a specific prefix (like `DRUPAL_`) they are injected as drupal config variables?

**Comment by seanhamlin** [comment]
Yeah, the code itself should not be too hard to write, this seems like an easy path forward. The only solution would be to write out a dynamic PHP include based on ENV VARs, but as you say, then we would be injecting code into the application.
